server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -0,0 +1,168 @@
+                    .description(e.getMessage())
+                    .build());
+        } catch (MailboxNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The mailbox '%s' was not found", mailboxId))
+                    .build());
+        } catch (MailboxParentNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The parent mailbox '%s' was not found.", e.getParentId()))
+                    .build());
+        } catch (MailboxHasChildException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description("Cannot update a parent mailbox.")
+                    .build());
+        } catch (MailboxException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("anErrorOccurred")
+                    .description("An error occurred when updating the mailbox")
+                    .build());
+        }
+    }
+
+    private Mailbox getMailbox(String mailboxId, MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) 
+            throws MailboxNotFoundException {
+
+        Optional<Mailbox> mailbox = mailboxUtils.mailboxFromMailboxId(mailboxId, mailboxSession);
+        if (!mailbox.isPresent()) {
+            throw new MailboxNotFoundException(mailboxId);
+        }
+        return mailbox.get();
+    }
+
+    private void ensureMailboxName(MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) {
+        if (updateRequest.getName().isPresent()) {
+            String name = updateRequest.getName().get();
+            char pathDelimiter = mailboxSession.getPathDelimiter();
+            if (name.contains(String.valueOf(pathDelimiter))) {
+                throw new MailboxNameException(String.format("The mailbox '%s' contains an illegal character: '%c'", name, pathDelimiter));
+            }
+        }
+    }
+
+    private void ensureParent(Mailbox mailbox, MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) 
+            throws MailboxException, MailboxHasChildException {
+
+        if (updateRequest.getParentId().isPresent()) {
+            String newParentId = updateRequest.getParentId().get();
+            Optional<MailboxPath> newParentMailboxPath = mailboxUtils.mailboxPathFromMailboxId(newParentId, mailboxSession);
+            if (!newParentMailboxPath.isPresent()) {
+                throw new MailboxParentNotFoundException(newParentId);
+            }
+            if (mailbox.getParentId().isPresent() && !mailbox.getParentId().get().equals(newParentId)
+                    && mailboxUtils.hasChildren(mailbox.getId(), mailboxSession)) {
+                throw new MailboxHasChildException();
```

Comment:
Ok, I get it ! You don't want to move a mailbox when it has children. So why bother with the first two conditions ?

(Note : I understand the second one as "Don't throw an exception if nothing changed". It should rather be "let's not do any computation if nothing changes, rather than modifying and making other conditions more complicated". No ?